msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignUpEmailOTPTest.kt

```diff
@@ -56,4 +68,158 @@ class SignUpEmailOTPTest : NativeAuthPublicClientApplicationAbstractTest() {
+            val user = config.email
+            val signUpResult = application.signUp(user)
+            Assert.assertTrue(signUpResult is SignUpError)
+            Assert.assertTrue((signUpResult as SignUpError).isUserAlreadyExists())
+        }
+    }
+
+    /**
+     * Sign up with email + OTP. Developer makes a request with invalid format email address.
+     * (use case 2.1.8, Test case 15)
+     */
+    @Test
+    fun testErrorInvalidEmailFormat() {
+        config = getConfig(defaultConfigType)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+            val user = INVALID_EMAIL
+            val signUpResult = application.signUp(user)
+            Assert.assertTrue(signUpResult is SignUpError)
+            Assert.assertTrue((signUpResult as SignUpError).isInvalidUsername())
+        }
+    }
+
+    /**
+     * Sign up with email + OTP. Developer can opt to get AT and/or ID token (aka sign in after signup).
+     * (use case 2.1.9, Test case 16)
+     */
+    @Test
+    fun testSignInAfterSignUp() {
+        config = getConfig(defaultConfigType)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        retryOperation {
+            runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+                val user = tempEmailApi.generateRandomEmailAddress()
+                val signUpResult = application.signUp(user)
+                assertResult<SignUpResult.CodeRequired>(signUpResult)
+                val otp = tempEmailApi.retrieveCodeFromInbox(user)
+                val submitCodeResult = (signUpResult as SignUpResult.CodeRequired).nextState.submitCode(otp)
+                assertResult<SignUpResult.Complete>(submitCodeResult)
+                val signWithContinuationResult = (submitCodeResult as SignUpResult.Complete).nextState.signIn()
+                assertResult<SignInResult.Complete>(signWithContinuationResult)
+            }
+        }
+    }
+
+    /**
+     * Sign up with email + OTP. Server requires password authentication, which is not supported by the developer (aka redirect flow).
+     * (use case 2.1.10, Test case 17)
+     */
+    @Test
+    fun testErrorRedirect() {
+        config = getConfig(ConfigType.SIGN_UP_PASSWORD)
+        application = setupPCA(config, listOf("oob"))
+
+        runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+            val user = tempEmailApi.generateRandomEmailAddress()
+            val signUpResult = application.signUp(user)
+            Assert.assertTrue(signUpResult is SignUpError)
+            Assert.assertTrue((signUpResult as SignUpError).isBrowserRequired())
+        }
+    }
+
+    /**
+     * Sign up with email + OTP. Server requires password authentication, which is supported by the developer.
+     * (hero scenario 11, use case 2.1.11, Test case 17)
```

Comment:
This should read "Test case 18"

Reply:
Actually not even sure test cases should be there, if you are referring to the ones in the test plan, E2E that were there previously have been removed so order has shifted. Use case is ok and enough to track down I would say.
An example is:
 * (hero scenario 9, use case 1.1.1,  Test case 13)
 Test case 13 is something else now, and Test Case 13 is twice in the code now

Reply:
Nice catch. Done.

Reply:
I agree. Test case encoding is more unreliable and unstable than user case. The original intention was to make it easier to track the progress of the test plan in the future, I will remove the test plan number, please let me know if you think it is inappropriate

Reply:
I would say remove the test case from all tests, even existing ones.

Reply:
Done.